src/pages/AddItem.js

```diff
@@ -14,24 +18,49 @@ export default function AddItem(props) {
     setPurchaseFrequency(e.target.value);
   };
 
-  const createListItem = (e) => {
-    e.preventDefault();
-    db.collection('generated_token')
-      .add({
-        item_name: itemName,
-        purchase_frequency: parseInt(purchaseFrequency),
-        last_purchased: lastPurchased,
-      })
-      .then((documentReference) => {
-        console.log('document reference ID', documentReference.id);
-      })
-      .catch((error) => {
-        console.log(error.message);
-      });
-    setItemName('');
-    setLastPurchased(null);
+  const doesItemExistInDatabase = () => {
+    const normalizedUserInput = itemName
+      .toLowerCase()
+      .replace(/[^\w]|_|\s/g, '');
+
+    const matchingItemName = props.listItems.filter((item) => {
+      const normalizedDatabaseItem = item.item_name
+        .toLowerCase()
+        .replace(/[^\w]|_|\s/g, '');
+      return normalizedDatabaseItem === normalizedUserInput;
+    });
+
+    return matchingItemName.length !== 0;
   };
 
+  function createListItem(e) {
+    e.preventDefault();
+
+    const newItemObject = {
+      item_name: itemName,
+      purchase_frequency: parseInt(purchaseFrequency),
+      last_purchased: lastPurchased,
+    };
+
+    const itemExists = doesItemExistInDatabase(itemName);
+
+    if (itemExists) {
+      swal(
+        'OH GOSH!',
+        `${itemName
+          .toUpperCase()
+          .replace(/[^\w\s]|_/g, '')} is already in your list`,
```

Comment:
Since this regex replace is used in multiple places, you may want to consider creating a helper function to do this. A thought for future refactoring.
```
const parseStr = (str) => {
  // define regex pattern in one place
  // do regex replace to str
  // return newStr
} 
```

Reply:
@yenly I agree! Great call creating a helper function. Here's what I came up with:

```javascript 
const normalizeString = (str) => {
   return str.toLowerCase().replace(/[^\w]|_|\s/g, '');
};
```
And then refactor `doesItemExistInDatabase()` to call `normalizeString()`

Reply:
Done!